### PR TITLE
Migration catalogue

### PR DESCRIPTION
**Pull request by NaryLozano**
Updated catalogue + Mocked responses

### CODE REVIEW

**Comment by NaryLozano** [comment]
Hello @xamcost I added the mocked data and updated the catalog to not make any requests.
Let me know if you have any comments and\or suggestions

**Review by xamcost — COMMENTED** [review]
Thank you for this work @NaryLozano ! It greatly simplifies the code base, making it more ready to welcome Sedimark catalogue once it will be available, while letting us carry our work with the mock catalogue you've set. I just left a few comments, no major structural changes. Let me know if you have any questions!

File: package-lock.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
         "react-jwt": "^1.2.0",
         "react-tag-input-component": "^2.0.2",
         "sharp": "^0.33.5",
+        "swr": "^2.2.5",
```
[the remark this answers is not in the document]
Comment (NaryLozano) [reply package-lock.json:21]: This should not be here. as we are not making any calls I deleted the use of SWR. following YAGNI

File: src/utils/data/mockCatalogue.json [hunk 2]
```diff
@@ -0,0 +1,673 @@
+{
```
[the remark this answers is not in the document]
Comment (NaryLozano) [reply src/utils/data/mockCatalogue.json:1]: Sure, It's just a prompt to `LeChat` from Mistral, given your examples for the [JSON](https://github.com/Sedimark/marketplace-frontend/issues/15#:~:text=%5B%0A%20%20%7B%22title%22,%5D%7D%2C%0A%20...%0A%5D) I asked it to generate similar ones

File: src/utils/data/mockProviders.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
+[
```
[the remark this answers is not in the document]
Comment (NaryLozano) [reply src/utils/data/mockProviders.json:1]: Oki, I was thinking the same, Once is needed I can add them no problem

File: src/app/catalogue/components/ResultsPane.js [hunk 4]
```diff
@@ -1,95 +1,139 @@
...
+   * @param {number} totalVcs - The total number of VCs.
+   * @param {number} batchSize - The batch size.
+   * @returns {number} The total number of pages.
+   */
+  function calculateTotalPages (totalVcs, batchSize) {
+    return Math.floor((totalVcs || 0) / batchSize)
+  }
+
+  const totalPages = useMemo( () => calculateTotalPages( totalVcs, settings.batchSize ), [totalVcs] )
+  const totalPagesToDisplay = totalPages
```
[the remark this answers is not in the document]
Comment (NaryLozano) [reply src/app/catalogue/components/ResultsPane.js:90]: so the pagination in this case was working along with the query made to the catalog, so its behavior is different now that it receives only one object with all the results.
+ You're right I was using Math.floor to round down to the nearest integer, but it actually needs to be rounded up 👍

**Comment by NaryLozano** [comment]
@xamcost Morning Maxime :)
Thank you for the feedback. I updated the files involved, and everything seems to be working fine. Let me know what you think!

File: package-lock.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (xamcost) [reply package-lock.json]: Perfect, that's better this way indeed. Thanks!

**Review by xamcost — APPROVED** [review]
Looks good to me! Thank you for your efforts @NaryLozano ! It works neatly. Feel free to merge anytime.

**Pull request merged**